internal/compiler/generator/rust.rs

```diff
@@ -374,6 +377,10 @@ fn generate_public_component(llr: &llr::PublicComponent) -> TokenStream {
                 slint::Weak::new(&self.0)
             }
 
+            fn as_component_rc(&self) -> vtable::VRc<slint::private_unstable_api::re_exports::ComponentVTable, Dyn> {
```

Comment:
Here it becomes visible how this new public API is exposing `private_unstable_api`. I still think that we may need a type-erased public type - a public wrapper around `VRc`. Or perhaps we should use the `Dyn` trick with `ComponentHandle` and `Weak`? (is that possible?)